x-pack/filebeat/input/http_endpoint/validate.go

```diff
@@ -5,15 +5,22 @@
 package http_endpoint
 
 import (
+	"crypto/hmac"
+	"crypto/sha1"
+	"encoding/hex"
 	"errors"
 	"fmt"
+	"io/ioutil"
 	"net/http"
 )
 
 type validator interface {
+	Validate(*http.Request) (int, error)
```

Comment:
To me it is a bit confusing to have an interface which defines methods that are meant to be called from eachother instead of by the user, since it is not evident when (or if) `validateHmac` and `validateHeaders` are called. Maybe changing validator to just have the `validate` method instead, and then either doing the validations inside without exposing `validateHmac` and `validateHeaders` through the interface, or having several validators, one for each validation and executing them in series would be a bit better. WDYT?

Reply:
Yeah I think that is a good assumption. At first I wanted to modify the withValidator func in handler.go to include just both validateHmac and validateHeaders, but if we ever want to add more validation steps I rather wanted to keep that part small, adding a generic validate method there instead. I think I can move the rest outside the interface for sure, still getting used to how golang uses them :)

Reply:
Header and body should be validated separately/differently. Processing the body might require some protecting from large contents as well (e.g. when sending an infinite stream using chunks).

I'd prefer to keep the method `ValidateHeader`, but change the interface to `ValidateHeader(http.Header)`, WDYT?